src/react-components/settings-menu.js

```diff
@@ -108,14 +123,21 @@ export default class SettingsMenu extends Component {
                   </i>
                 </div>
                 <div className={styles.listItem}>
-                  <StateLink
-                    stateKey="modal"
-                    stateValue="close_room"
-                    history={this.props.history}
-                    onClick={this.props.hideSettings}
+                  <a
+                    href="#"
+                    onClick={() =>
+                      this.props.performConditionalSignIn(
+                        () => this.props.hubChannel.can("update_hub"),
+                        () => {
+                          this.props.pushHistoryState("modal", "close_room");
```

Comment:
Aside: I'm increasingly inclined to stick functions like `performConditionalSignIn` and `pushHistoryState` into our `APP` global (under appropriate namespaces) since it's getting a bit silly to pass them along in props, like we do, as if they are ever going to mean different things to different react components.

Reply:
I guess React has Contexts to solve this. TIL the problem is common enough to have been coined "prop drilling".

Reply:
ah yeah I agree a Context seems a good choice for a few of the things we have. i think it probably deserves its own refactor PR vs me just adding this one function, since I think there are several candidates.